src/file/file.service.ts

```diff
@@ -18,7 +19,7 @@ export class FileService {
 
       return fs.createReadStream(file);
     } else if (file.startsWith('http')) {
-      const content = this.cloudProviders.get(file);
+      const content = await this.cloudProviders.get(file) || await axios(file).then(({ data }) => data);
```

Comment:
Don't mix `Promise` API and async/await. Please rework it to make code more clean